testdata/p4_16_samples/psa-portid-using-newtype2.p4

```diff
@@ -0,0 +1,293 @@
+#include "core.p4"
+
+// From psa.p4
+
+#define PORT_BITWIDTH 9  // target-specific
+
+// PortIdUInt_t is only intended for casting values from type PortId_t
+// to an unsigned integer with the same width.
+typedef bit<PORT_BITWIDTH> PortIdUInt_t;
+
+// request translation by P4Runtime agent of all occurrences of PortId_t exposed
+// to control-plane
+@p4runtime_translate("port")
+newtype bit<PORT_BITWIDTH> PortId_t;
```

Comment:
I do not understand why the annotation is required here. Again, maybe I mis-understand the feature, but at least my vision for it was that we are basically saying that variables, that are declared as PortId_t in the P4 program will behave as `bit<PORT_BITWIDTH>` in the data plane program, but will be exposed in the control plane as `control_plane_specific_representation_of(PortId_t)`. 

This "specific representation" consists of (for example):
 - The language-specific type that the control-plane will use for PortId_t (e.g. uint32_t regardless of whether the base type is bit<7> or <bit<9> or bit <17>)
- The control-plane-specific representation format of the underlying values (e.g. an integer (0, 1, 2, 3), or a formatted string such as "eth0", "port-3/5" or anything else)
- The control-plane-specific rules to convert from the control-plane-specific representation of the values into the underlying data plane type (bit<PORT_BITWIDTH>) and vice-versa.

If we agree on that, then there is no need to add the annotation, because translation falls in the category of the possible conversion rules. The data plane program still is unaware of that and can't do much about it (and it should not).

Reply:
I think you are relying on a reserved type name: `PortId_t`. That's fine. I think @jafingerhut does not - so he wants to identify the interesting types somehow.

Reply:
PortId_t is just one of 3 types defined in PSA that has target-specific bit widths, each with a different numerical translation we expect the agent to be able to do: PortId_t, ClassOfService_t, and Timestamp_t.

There are 4 more types defined in PSA that we want target-specific bit widths, but no numerical translation in the agent: MulticastGroup_t, CloneSessionId_t, PacketLength_t, EgressInstance_t

You have mentioned in the meetings types like IPv4Address, MACAddress, IPv6Address, where you would like the control plane to know about these types in a P4 program, so it can provide custom debug and/or string<->numeric value conversions.

This set of types will only grow in the future, yes?

By what mechanism do you propose to inform the compiler about the properties of these types, if not annotations on the 'newtype' or 'typedef' definition of the type?

Reply:
Yes. This is correct.

Reply:
Given that a P4Runtime server will likely need to have at least a little bit of PSA-specific code in it to work with a PSA target, whether it has PSA-specific code to recognize a list of type names like `PortId_t` and `ClassOfService_t`, or annotation keys with specific names, seems to be six of one vs. half a dozen of another.  Both seem like they could be made to work about as easily as each other, and I can see reasons why one might prefer to identify these types by type names like `PortId_t` rather than an annotation key name.